packages/fxa-auth-server/lib/subhub.js

```diff
@@ -0,0 +1,315 @@
+          product_id: isA.string().required(),
+          interval: isA.string().required(),
+          amount: isA.number().required(),
+          currency: isA.string().required()
+        }))
+      }
+    },
+
+    listSubscriptions: {
+      path: '/customer/:uid/subscriptions',
+      method: 'GET',
+      validate: {
+        params: {
+          uid: isA.string().required(),
+        },
+        // TODO: update with final subscriptions schema from subhub
+        response: isA.array().items(isA.object({
+          plan_id: isA.string().required(),
+          product_id: isA.string().required(),
+          current_period_end: isA.number().required(),
+          end_at: isA.number().required(),
+        }))
+      }
+    },
+
+    getCustomer: {
+      path: '/customer/:uid',
+      method: 'GET',
+      validate: {
+        params: {
+          uid: isA.string().required(),
+        },
+        // TODO: update with final customer schema from subhub
+        response: isA.object()
+      }
+    },
+
+    updateCustomer: {
+      path: '/customer/:uid',
+      method: 'POST',
+      validate: {
+        params: {
+          uid: isA.string().required(),
+        },
+        payload: {
+          pmt_token: isA.string().required(),
+        },
+        response: isA.alternatives(
+          // TODO: update with final customer schema from subhub
+          isA.object(),
+          isA.object({
+            message: isA.string()
+          })
+        )
+      }
+    },
+
+    createSubscription: {
+      path: '/customer/:uid/subscriptions',
+      method: 'POST',
+      validate: {
+        params: {
+          uid: isA.string().required(),
+        },
+        payload: {
+          pmt_token: isA.string().required(),
+          plan_id: isA.string().required(),
+          email: isA.string().required(),
+        },
+        response: isA.alternatives(
+          isA.object({
+            sub_id: isA.string()
+          }),
+          isA.object({
+            message: isA.string()
+          })
+        )
+      }
+    },
+
+    cancelSubscription: {
+      path: '/customer/:uid/subscriptions/:sub_id',
+      method: 'DELETE',
+      validate: {
+        params: {
+          uid: isA.string().required(),
+          sub_id: isA.string().required(),
+        },
+        response: isA.alternatives(
+          isA.object({}),
+          isA.object({
+            message: isA.string()
+          })
+        )
+      }
+    },
+
+  });
+
+  const api = new SubHubAPI(
+    config.subhub.url,
+    {
+      headers: {
+        // TODO: update with subhub final auth
+        Authorization: `Bearer ${config.subhub.key}`
+      },
+      timeout: 15000
+    }
+  );
+
+  return {
+    isStubAPI: false,
+
+    async listPlans() {
+      return api.listPlans();
+    },
+
+    async listSubscriptions(uid) {
+      try {
+        return await api.listSubscriptions(uid);
+      } catch (err) {
+        if (err.statusCode === 404) {
```

Comment:
Is this status code guaranteed to be fixed? Is a status code/message combo how subhub defines their errors?

Reply:
The answers to both of these are 🤷‍♂️ TBD. This part of the subhub spec is not yet finished.

Reply:
Yeah, I think this is a part of issue #719